### PR TITLE
ci: list_all_builtin_functions_ordered.

### PR DESCRIPTION
**Pull request by youngsofun**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

Closes #issue

### CODE REVIEW

File: src/query/functions/tests/it/scalars/mod.rs [hunk 1]
```diff
@@ -252,6 +252,42 @@ fn list_all_builtin_functions() {
...
+fn list_all_builtin_functions_with_ordered() {
+    let mut mint = Mint::new("tests/it/scalars/testdata");
+    let file = &mut mint.new_goldenfile("function_list_ordered.txt").unwrap();
+
+    let fn_registry = &BUILTIN_FUNCTIONS;
+
+    let mut funcs = fn_registry
+        .funcs
+        .iter()
+        .flat_map(|(_name, funcs)| funcs)
```
Comment (andylokandy) [inline src/query/functions/tests/it/scalars/mod.rs:265]:
```suggestion
        .values()
        .flat_map(|funcs| funcs)
```

File: src/query/functions/tests/it/scalars/mod.rs [hunk 2]
```diff
@@ -252,6 +252,42 @@ fn list_all_builtin_functions() {
...
+            funcs
+                .iter()
+                .map(|(_, seq)| ((name.clone(), *seq), "FACTORY".to_string()))
+        })
+        .collect::<Vec<_>>();
+
+    funcs.extend(facts);
+    funcs.sort();
+    for ((name, seq), sig) in funcs {
+        writeln!(file, "{name} {seq} {sig}").unwrap();
```
Comment (andylokandy) [inline src/query/functions/tests/it/scalars/mod.rs:287]: The function name is duplicated. Is this sufficient?

```suggestion
        writeln!(file, "{seq} {sig}").unwrap();
```

**Review by andylokandy — APPROVED** [review]
rest LGTM

File: src/query/functions/tests/it/scalars/mod.rs (hunk 2 above)
Comment (youngsofun) [reply src/query/functions/tests/it/scalars/mod.rs:287]: factory has only name, sig==""

**Comment by youngsofun** [comment]
ready,
but better merge after https://github.com/datafuselabs/databend/pull/10343

**Review by sundy-li — APPROVED** [review]

**Pull request merged**